homeassistant/components/media_player/sonos.py

```diff
@@ -472,6 +477,44 @@ def update(self):
                     support_next_track = True
                     support_pause = True
 
+                    position_info = self._player.avTransport.GetPositionInfo(
+                        [('InstanceID', 0),
+                         ('Channel', 'Master')]
+                    )
+                    rel_time = _parse_timespan(
+                        position_info.get("RelTime")
+                    )
+
+                    # player no longer reports position?
+                    update_media_position = rel_time is None and \
+                        self._media_position is not None
+
+                    # player started reporting position?
+                    update_media_position |= rel_time is not None and \
+                        self._media_position is None
+
+                    # position changed?
+                    if rel_time is not None and \
+                       self._media_position is not None:
+
+                        calculated_position = \
+                            self._media_position + \
+                            time.time() - \
+                            self._media_position_updated_at
+
+                        update_media_position = \
+                            abs(calculated_position - rel_time) > 1.5
+
+                    if update_media_position:
+                        media_position = rel_time
+                        media_position_updated_at = time.time()
```

Comment:
Please use `homeassistant.util.dt.utcnow()`, as that's the date time type that we use throughout Home Assistant.